app/src/main/java/ch/epfl/qedit/view/QR/ScannerActivity.java

```diff
@@ -0,0 +1,119 @@
+import androidx.core.content.ContextCompat;
+import ch.epfl.qedit.view.util.ConfirmDialog;
+import com.google.zxing.Result;
+import me.dm7.barcodescanner.zxing.ZXingScannerView;
+
+/** * this class is largely inspired from priyanka pakhale work available on github ** */
+public class ScannerActivity extends AppCompatActivity
+        implements ZXingScannerView.ResultHandler, ConfirmDialog.ConfirmationListener {
+
+    private static final int REQUEST_CAMERA = 1;
+    private ZXingScannerView scannerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        scannerView = new ZXingScannerView(this);
+        setContentView(scannerView);
+        int currentApiVersion = Build.VERSION.SDK_INT;
+
+        if (currentApiVersion >= Build.VERSION_CODES.M) {
+            if (checkPermission()) {
+                Toast.makeText(
+                                getApplicationContext(),
+                                "Permission already granted!",
+                                Toast.LENGTH_LONG)
+                        .show();
+            } else {
+                requestPermission();
+            }
+        }
+    }
+
+    private boolean checkPermission() {
+        return (ContextCompat.checkSelfPermission(getApplicationContext(), CAMERA)
+                == PackageManager.PERMISSION_GRANTED);
+    }
+
+    private void requestPermission() {
+        ActivityCompat.requestPermissions(this, new String[] {CAMERA}, REQUEST_CAMERA);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (checkPermission()) {
+                checkNullScanner();
+                scannerView.setResultHandler(this);
+                scannerView.startCamera();
+            } else {
+                requestPermission();
+            }
+        }
+    }
+
+    public void checkNullScanner() {
+        if (scannerView == null) {
+            scannerView = new ZXingScannerView(this);
+            setContentView(scannerView);
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        scannerView.stopCamera();
+    }
+
+    public void onRequestPermissionsResult(
+            int requestCode, String permissions[], int[] grantResults) {
+        if (requestCode == REQUEST_CAMERA && grantResults.length > 0) {
+
+            boolean cameraAccepted = grantResults[0] == PackageManager.PERMISSION_GRANTED;
+            if (cameraAccepted) {
+                Toast.makeText(this, "Permission Granted", Toast.LENGTH_LONG).show();
+            } else {
+                Toast.makeText(this, "Permission Denied", Toast.LENGTH_LONG).show();
+                super.onBackPressed();
+            }
+        }
+    }
+
+    @Override
+    public void handleResult(Result result) {
+        /*Log.d("QRCodeScanner", result.getText());
```

Comment:
Can you make sure to remove the debug comments before merging ? thanks !

Reply:
Yep